### DISCUSSION TITLE
How to create yup validator with array of string type for checkboxes

### DISCUSSION THREAD

**Discussion by arpitdalal** [airjp73/rvf#101] [Q&A] [answered]
I'm using `withYup` and `yup` and I've this as my validator:
```tsx
const validator = withYup(
  yup.object({
    checkboxes: yup.array(yup.string()),
  }),
);
```
but when I submit my checkboxes using this validator, I face a weird behavior/bug. If only 1 checkbox is submitted, the `validator.validate` actually returns a string type instead of an array for some reason but when submitting multiple checkboxes (2 or more), `validator.validate` gives the correct `array` type.
NOTE: all the checkboxes have the same name

Please see this example on [stackblitz](https://stackblitz.com/github/arpitdalal/rmx-validated-form-checkbox) and [github repo](https://github.com/arpitdalal/rmx-validated-form-checkbox)

Is my `validator` wrong for this use case or is this a bug? Because I believe, `validator.validate` should give me the same object as the original `formData`

**Comment by arpitdalal** [comment]
A little more digging tells me that there is something wrong in [flatten.ts](https://github.com/airjp73/remix-validated-form/blob/main/packages/remix-validated-form/src/internal/flatten.ts) or [MultiValueMap.ts](https://github.com/airjp73/remix-validated-form/blob/main/packages/remix-validated-form/src/internal/MultiValueMap.ts) which is flattening the one value array

**Comment by arpitdalal** [comment]
New [issue ](https://github.com/airjp73/remix-validated-form/issues/102)created
Edit: because I believe this is a bug in the lib itself

**Comment by airjp73** [comment, marked as answer]
I responded on the issue, but responding here as well for completeness. Unfortunately, this behavior is a known and [documented](https://www.remix-validated-form.io/repeated-field-names) limitation. To support this use case, you need to write a validator that can handle it being either a single string or an array of strings. I believe there's a way to preprocess values in yup that you could use to do this, but I don't know how that works off the top of my head.
